src/db/notifications/listen.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @fileOverview
+ *
+ * @see https://github.com/vitaly-t/pg-promise/wiki/Robust-Listeners
+ */
+const config = require('config');
+const logger = require('../../tools/logger');
+const Promise = require('bluebird');
+const pgPromise = require('pg-promise');
+
+const pgp = pgPromise();
+const db = pgp(config.get('db'));
+
+async function reconnect({ delay = 1000, maxAttempts = 1, onLost }) {
+  const retryAttemptsLeft = maxAttempts - 1;
+
+  try {
+    const connection = await db.connect({ direct: true, onLost });
```

Comment:
btw, existing reconnect wasn't working

source of issue was reconnection line of code
```
try {
return db.connect(...)
} catch (err) {

}
```
would never get to catch block since once function connect is called promise is returned, so, functions execution succeeds.
Promise got rejected outside of `reconnect` function and so retry is never called.

Right now it works fine. Tested locally by shutting down db server manually.